### ISSUE TITLE
Possible bug in zero_center_normalize with method = "GMM"?

### ISSUE THREAD

**Issue by rohan-shah** [statisticalbiotechnology/diffacto#9]
I've been looking at the GMM normalization method in diffacto, and think there might be a bug. The relevant code is:
```
        ''' two-component Gaussian mixture model '''
        from sklearn.mixture import GMM
        gmm = GMM(2)
        norm_scale = []
        for sp in samples:
            v = df[sp].values
            v = v[np.logical_not(np.isnan(v))]
            v = v[np.logical_not(np.isinf(v))]
            try:
                gmm.fit(np.matrix(v.values).T) #no property .values, so bug?
                vmean = gmm.means_[np.argmin(gmm.covars_)]
                norm_scale.append(vmean)
            except:
                norm_scale.append(np.nanmean(v))
        norm_scale = np.array(norm_scale)
```
Notice that v is set to a numpy array, but then in the try block "v.values" is used. In my debugging so far, v has never had a values property, resulting in an exception, so the code calls np.nanmean instead. Is this a bug?

**Comment by userbz** [comment]
Yes. Seemed that .values is called twice, and the second time should be a bug.
